harmony/harmony.py

```diff
@@ -209,18 +211,49 @@ def _temporal_subset_params(self, request: Request) -> list:
         else:
             return []
 
-    def submit(self, request: Request) -> Optional[dict]:
+    def submit(self, request: Request) -> Optional[str]:
         """Submits a request to Harmony and returns the Harmony job details."""
         if not request.is_valid():
             msgs = ', '.join(request.error_messages())
             raise Exception(f"Cannot submit an invalid request: [{msgs}]")
 
-        job = None
+        job_id = None
         session = self._session()
-        response = session.get(self._url(request), params=self._params(request)).result()
+        response = session.get(self._submit_url(request), params=self._params(request)).result()
         if response.ok:
-            job = response.json()
+            job_id = (response.json())['jobID']
         else:
             response.raise_for_status()
 
-        return job
+        return job_id
+
+    def status(self, job_id: str, progress_only=False) -> Union[str, dict]:
+        """Returns job metadata from Harmony.
+
+        If ``progress_only`` is True, return the job's percent complete. If ``progress_only`` is
+        False, return dict containing job metadata.
+
+        Args:
+            job_id: UUID string for the job you wish to interrogate.
+            progress_only: An option to change the output format.
+
+        Returns:
+            Either the percent complete or a dict of metadata.
+
+        :raises Exception: This can happen if an invalid job_id is provided or Harmony services
+        can't be reached.
+        """
+        session = self._session()
+        response = session.get(self._status_url(job_id)).result()
+        if response.ok:
+            if progress_only:
+                return (response.json())['progress']
+            else:
+                fields = [
+                    'status', 'message', 'progress', 'createdAt', 'updatedAt', 'request',
```

Comment:
Should we convert str -> datetime for createdAt and updatedAt?

Reply:
I can do that.

Reply:
Note that I've chosen to import another third party library to handle ISO 8601 datetimes:

https://stackoverflow.com/questions/969285/how-do-i-translate-an-iso-8601-datetime-string-into-a-python-datetime-object
"As from python 3.7 you can use datetime.datetime.fromisoformat" but "The official documentation from the link you gave reads the following caution: This does not support parsing arbitrary ISO 8601 strings - it is only intended as the inverse operation of datetime.isoformat(). A more full-featured ISO 8601 parser, dateutil.parser.isoparse is available in the third-party package dateutil. So yeah, even for Python 3.7 we are back to the dateutil package."

I verified datetime.datetime.fromisoformat() is unable to parse an example Harmony-generated datetime '2021-02-25T19:18:39.798Z'. I was able to parse the datetime string with:

import dateutil.parser
dateutil.parser.parse('2021-02-25T19:18:39.798Z')

Reply:
Hmm - looks like that was already satisfied by another import but I added it to `requirements/core.txt`.